delta/sourcing-psql/src/main/scala/ch/epfl/bluebrain/nexus/delta/sourcing/ScopedEventLog.scala

```diff
@@ -225,14 +227,20 @@ object ScopedEventLog {
 
       override def evaluate(ref: ProjectRef, id: Id, command: Command): IO[(E, S)] = {
 
-        def saveTag(event: E, state: S): IO[ConnectionIO[Unit]] =
-          tagger.tagWhen(event).fold(IO.pure(noop)) { case (tag, rev) =>
-            if (rev == state.rev)
-              IO.pure(stateStore.save(state, tag, Noop))
-            else
+        def newTaggedState(event: E, state: S): IO[Option[(UserTag, S)]] =
+          tagger.tagWhen(event) match {
+            case Some((tag, rev)) if rev == state.rev =>
+              IO.some(tag -> state)
+            case Some((tag, rev))                     =>
               stateMachine
                 .computeState(eventStore.history(ref, id, Some(rev)))
-                .map(_.fold(noop) { s => stateStore.save(s, tag, Noop) })
+                .flatTap {
+                  case stateOpt if !stateOpt.exists(_.rev == rev) =>
```

Comment:
To clarify, the error should be raised even when this returns `None`? Instead of a `noop`

Reply:
If it is none or if for some reason, the state could not be replayed up to the requested rev